File: GGC-Mobile/src/entities/Evenement.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package entities;

import java.sql.Date;
import java.time.LocalDate;
import javafx.scene.control.Button;

/**
 *
 * @author Azer Lahmer
 */
public class Evenement {

    private int reference;
    private Date dateDebut;
    private Date dateFin;
    //private String heureDebut;
    //private String heureFin;
    private String localisation;
    private String description;
    private int nbrParticipant;
    private String Titre; 
    private Button participer;

    public Evenement() {
    }

    public Evenement(int reference) {
        this.reference = reference;
    }

    public Evenement(int reference, Date dateDebut, Date dateFin, String localisation, String description, int nbrParticipant, String Titre) {
        this.reference = reference;
        this.dateDebut = dateDebut;
        this.dateFin = dateFin;
        this.localisation = localisation;
        this.description = description;
        this.nbrParticipant = nbrParticipant;
        this.Titre = Titre;
    }

    public String getTitre() {
        return Titre;
    }

    public void setTitre(String Titre) {
        this.Titre = Titre;
    }



    public int getReference() {
        return reference;
    }

    public Date getDateDebut() {
        return dateDebut;
    }

    /*public String getHeureDebut() {
        return heureDebut;
    }

    public String getHeureFin() {
        return heureFin;
    }*/

    public String getLocalisation() {
        return localisation;
    }

    public String getDescription() {
        return description;
    }

    public int getNbrParticipant() {
        return nbrParticipant;
    }

//    public String getImage() {
//        return image;
//    }

//    public void setImage(String image) {
//        this.image = image;
//    }
    

    public void setReference(int reference) {
        this.reference = reference;
    }

    public void setDateDebut(Date dateDebut) {
        this.dateDebut = dateDebut;
    }

    public void setDateFin(Date dateFin) {
        this.dateFin = dateFin;
    }

    public Date getDateFin() {
        return dateFin;
    }

   /* public void setHeureDebut(String heureDebut) {
        this.heureDebut = heureDebut;
    }

    public void setHeureFin(String heureFin) {
        this.heureFin = heureFin;
    }*/

    public void setLocalisation(String localisation) {
        this.localisation = localisation;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public void setNbrParticipant(int nbrParticipant) {
        this.nbrParticipant = nbrParticipant;
    }

    @Override
    public String toString() {
        return "Evenement{" + "reference=" + reference + ", dateDebut=" + dateDebut + ", dateFin=" + dateFin  + ", localisation=" + localisation + ", description=" + description + ", nbrParticipant=" + nbrParticipant + '}';
    }

}

File: GGC-WEB/src/Entity/Evenement.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;
use Symfony\Bridge\Doctrine\Validator\Constraints\UniqueEntity;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Symfony\Component\Serializer\Annotation\Groups;

/**
 * Evenement
 *
 * @ORM\Table(name="evenement")
 * @ORM\Entity(repositoryClass="App\Repository\EvenementRepository")
 * @UniqueEntity(fields="reference", message="Un evenement existe déjà avec cette reference.")
 */
class Evenement
{
    /**
     * @var int
     *
     * @ORM\Column(name="reference", type="integer", nullable=false)
     * @ORM\Id
     * @Groups("event:read")
     */
    private $reference;

    /**

     * @ORM\Column(name="dateDebut", type="date", nullable=false)
     * @Assert\GreaterThanOrEqual("today",message="La date du debut doit être supérieure à la date d'aujourd'hui"))
     * @Groups("event:read")
     */
    private $datedebut;

    /**
     
     *
     * @ORM\Column(name="dateFin", type="date", nullable=false)
     * @Assert\GreaterThanOrEqual(propertyPath="dateDebut",
    * message="La date du fin doit être supérieure à la date début")
     * @Groups("event:read")
     */
    private $datefin;

    /**
     * @var string
     *
     * @ORM\Column(name="localisation", type="string", length=50, nullable=false)
         * @Assert\Length(
     *      min = 3,
     *      max = 100,
     *      minMessage = "Your first name must be at least {{ limit }} characters long",
     *      maxMessage = "Your first name cannot be longer than {{ limit }} characters"
     * )
     * @Groups("event:read")
     */
    private $localisation;

    /**
     * @var string
     *
     * @ORM\Column(name="description", type="string", length=50, nullable=false)
     *  * @Assert\Length(
     *      min = 10,
     *      max = 250,
     *      minMessage = "Your first name must be at least {{ limit }} characters long",
     *      maxMessage = "Your first name cannot be longer than {{ limit }} characters")
     * @Groups("event:read")
     */
    private $description;
     /**
     * @var string
     *
     * @ORM\Column(name="photo", type="string", length=50, nullable=true)
       * @Assert\NotBlank(message="Please, upload the photo.")
     *  @Assert\File(mimeTypes={ "image/png", "image/jpeg" , "image/jpg" })
      * @Groups("event:read")
     */
    private $photo;

    /**
     * @var int
     *
     * @ORM\Column(name="nbrParticipant", type="integer", nullable=false)
     * @Assert\Positive(message="le nbr Participants doit etre positive")
     * @Groups("event:read")
     */
    private $nbrparticipant;

    /**
     * @var string
     *
     * @ORM\Column(name="Titre", type="string", length=50, nullable=false)
     * @Groups("event:read")
     */
    private $titre;

    public function getReference(): ?int
    {
        return $this->reference;
    }

    public function getDatedebut(): ?\DateTimeInterface
    {
        return $this->datedebut;
    }

    public function setDatedebut(\DateTimeInterface $datedebut): self
    {
        $this->datedebut = $datedebut;

        return $this;
    }

    public function getDatefin(): ?\DateTimeInterface
    {
        return $this->datefin;
    }

    public function setDatefin(\DateTimeInterface $datefin): self
    {
        $this->datefin = $datefin;

        return $this;
    }

    public function getLocalisation(): ?string
    {
        return $this->localisation;
    }

    public function setLocalisation(string $localisation): self
    {
        $this->localisation = $localisation;

        return $this;
    }

    public function setReference(int $reference): self
    {
        $this->reference = $reference;

        return $this;
    }
    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(string $description): self
    {
        $this->description = $description;

        return $this;
    }
    public function getPhoto()
    {
        return $this->photo;
    }

    public function setPhoto($photo)
    {
        $this->photo = $photo;

        return $this;
    }

    public function getNbrparticipant(): ?int
    {
        return $this->nbrparticipant;
    }

    public function setNbrparticipant(int $nbrparticipant): self
    {
        $this->nbrparticipant = $nbrparticipant;

        return $this;
    }

    public function getTitre(): ?string
    {
        return $this->titre;
    }

    public function setTitre(string $titre): self
    {
        $this->titre = $titre;

        return $this;
    }

 /**
     * @return Collection|Participation[]
     */
    public function getParticipation(): Collection
    {
        return $this->participation;
    }

    /**
     * @ORM\OneToMany(targetEntity=Participation::class, mappedBy="idEvent",cascade={"remove"}, orphanRemoval=true )
     */
    private $participation;

    public function __construct2()
    {
        $this->participation = new ArrayCollection();
    }
    public function __toString()
    {
        return (string)$this->getReference();
    }
}
